From d484bef133af9c87d64899fc1e1d0be2a7c7785b Mon Sep 17 00:00:00 2001
From: Phil Elwell <phil@raspberrypi.com>
Date: Mon, 20 Nov 2023 10:16:10 +0000
Subject: [PATCH] overlays: ov5647: cam0 mode should use cam0_reg

When the cam0 parameter is used, the vcm should be updated to refer to
the cam0 regulator.

See: https://github.com/raspberrypi/linux/issues/5722

Signed-off-by: Phil Elwell <phil@raspberrypi.com>
---
 arch/arm/boot/dts/overlays/ov5647-overlay.dts | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

--- a/arch/arm/boot/dts/overlays/ov5647-overlay.dts
+++ b/arch/arm/boot/dts/overlays/ov5647-overlay.dts
@@ -77,7 +77,8 @@
 		       <&reg_frag>, "target:0=",<&cam0_reg>,
 		       <&clk_frag>, "target:0=",<&cam0_clk>,
 		       <&cam_node>, "clocks:0=",<&cam0_clk>,
-		       <&cam_node>, "avdd-supply:0=",<&cam0_reg>;
+		       <&cam_node>, "avdd-supply:0=",<&cam0_reg>,
+		       <&vcm_node>, "VANA-supply:0=",<&cam0_reg>;
 		vcm = <&vcm_node>, "status=okay",
 		       <&cam_node>,"lens-focus:0=", <&vcm_node>;
 	};
